spec/datadog/appsec/component_spec.rb

```diff
@@ -59,7 +59,112 @@
     end
   end
 
+  describe '#reconfigure' do
+    context 'lock' do
+      it 'makes sure to synchronize' do
+        mutex = Mutex.new
+        processor = instance_double(Datadog::AppSec::Processor)
+        component = described_class.new(processor: processor, mutex: mutex)
+        expect(mutex).to receive(:synchronize)
```

Comment:
I had to tweak your suggestion to avoid messing up with Rspec internal `Mutex`

```ruby
it 'makes sure to synchronize' do
  mutex = Mutex.new
  processor = instance_double(Datadog::AppSec::Processor)
  component = described_class.new(processor: processor)
  component.instance_variable_set(:@mutex, mutex)
  expect(mutex).to receive(:synchronize)
  component.reconfigure(ruleset: {})
end
```